lib/Parse/ParseExpr.cpp

```diff
@@ -2013,8 +2012,8 @@ ParserResult<Expr> Parser::parseExprStringLiteral() {
     Status = parseStringSegments(Segments, EntireTok, InterpolationVar, 
                                  Stmts, LiteralCapacity, InterpolationCount);
 
-    auto Body = BraceStmt::create(Context, Loc, Stmts, EndLoc,
-                                  /*implicit=*/false);
+    auto Body = BraceStmt::create(Context, Loc, Stmts, /*endLoc=*/Loc,
+                                  /*implicit=*/true);
```

Comment:
@brentdax Was there a reason *not* to make it `implict`?

Reply:
I honestly don't remember at this point. In general, the scoping behavior of `InterpolatedStringLiteralExpr` is fairly brittle. I would verify that this not only doesn't regress any tests normally, but also with ASTScope enabled.

(I might also check if changing to the closing brace to `EndLoc` allows you to remove [this hack](https://github.com/apple/swift/blob/master/lib/Sema/TypeCheckCaptures.cpp#L217-L223); it's entirely possible that I did the wrong thing here.)

Reply:
It looks like the hack is still needed - it fails to build the stdlib if I comment it out.
@brentdax or @davidungar how do I run the tests with ASTScope enabled? I tried forcing `EnableASTScope` to `true` in CompilerInvocation.cpp, but that also caused the stdlib build to fail.

Reply:
Looks like you want to run `env SWIFT_TEST_OPTIONS='-enable-astscope-lookup' <test command here>`, assuming you're running tests either directly or indirectly through lit.

My understanding is that ASTScope needs the BraceStmt's EndLoc to be after all of the interpolations that might occur in the string literal, but @davidungar knows much more about that than I do.

Reply:
Thanks @brentdax! I ran `build-script` with `-t` and `SWIFT_TEST_OPTIONS` locally both with and without the changes in this PR and it looks like it improves things. I get 1132 unexpected failures without the changes and 1118 with them.

@davidungar Let me know if you think this will cause any problems with ASTScope. If it uses Lexer::getCharSourceRangeFromSourceRange(...) to get the character-based CharSourceRange of AST nodes from the token-start-position-based StartLoc/EndLoc/SourceRange they store, I don't think it'll need to do any special-casing for interpolated string literals or their implicit children after this change.